pkg/kds/v2/client/kds_client.go

```diff
@@ -75,6 +87,16 @@ func (s *kdsSyncClient) Receive() error {
 			}
 			return errors.Wrap(err, "failed to receive a discovery response")
 		}
+
+		if err := received.Validate(); err != nil {
+			if err := s.kdsStream.NACK(received.Type, err); err != nil {
```

Comment:
please restore the log `s.log.Info("error during callback received, sending NACK", "err", err)`, it's useful for debugging

Reply:
ah but the text should be different